### DISCUSSION TITLE
`ui/embed.go:10:12: pattern all:dist: no matching files found` error when running "go get" or building project

### DISCUSSION THREAD

**Discussion by janzheng** [pocketbase/pocketbase#46] [Q&A] [answered]
Hi there,

I'm completely new to Go, and trying to run the project locally, or doing a `go get github.com/pocketbase/pocketbase` but running into the error: `../../../go/pkg/mod/github.com/pocketbase/pocketbase@v0.1.2/ui/embed.go:10:12: pattern all:dist: no matching files found`

Not sure if this is problem with the package, my system, or how I'm running it?

Thanks!!

**Comment by ganigeorgiev** [comment, marked as answer]
To run PocketBase locally, you don't have to install anything - just download the [latest release](https://github.com/pocketbase/pocketbase/releases) for your platform, and run the binary with `./pocketbase serve`.

PocketBase is primarily a Go package so on its own you cannot run it directly. To build the base PocketBase application locally:
- download the repo
- navigate to the [`examples/base`](https://github.com/pocketbase/pocketbase/tree/master/examples/base) directory
- run `go build`

To run the generated binary, just execute it with the serve argument, eg. `./base serve`.

**Reply by janzheng** [reply]
Thanks so much! I was trying to compile it and deploy it on Railway.app, but instead went the binary route and deployed on Digital Ocean and it works so well!

For anyone who's very new to all this stuff and wants to deploy on Digital Ocean, I did the following to deploy a live service on DO:

1. Sign up on Digital Ocean
2. Get the cheapest Droplet ($4/mo) to run Ubuntu
3. Download the Ubuntu binary from Pocketbase
4. Upload the binary to the Droplet using SFTP, by opening up console and doing `sftp root@droplet-ip-address`, then `put /location/of/binary`
5. Then exit sftp and SSH into the droplet with `ssh root@droplet-ip-address`
6. Set up the firewall with:
```
ufw allow 443
ufw allow 80
```
7. Get a domain set up, by using one of my existing domains (on Cloudflare) and point the A record of pocket.mydomain.com to the IP address of the Droplet (this is necessary for https / cert to work)
8. Then you can check to start the app with `./pocketbase serve --http="pocket.mydomain.com:80" --https="pocket.mydomain.com:443"`

However, to keep it running without stopping after you log out of SSH, you:
a. ssh and run the command `./pocketbase serve --http="pocket.mydomain.com:80" --https="pocket.mydomain.com:443"`
b. hit ctrl z to suspend the service
c. type `bg` and hit enter to send it to background
d. type `lsof -i` and hit enter to get the pid of the pocketbase service
e. type `disown [pid]` and hit enter — disown makes pocketbase run even after logging out of SSH

To kill the service, just type `kill [pid]`

Hope this helps others!

**Reply by ganigeorgiev** [reply]
Thanks @janzheng for the instructions!

btw. To keep the service running, even if you restart the server or the process get killed, you may also create a systemd service, as shown in the ["Going to production" recommendation guide](https://pocketbase.io/docs/going-to-production/#create-systemd-service)

**Comment by jimafisk** [comment]
If you are trying to build from source instead of downloading a release, an error like `pattern all:dist: no matching files found` seems to me like you're using an older version of Go that doesn't support `all:pattern` in the [embed pkg](https://pkg.go.dev/embed), which was added in v1.18: https://go-review.googlesource.com/c/website/+/393234/

**Reply by ganigeorgiev** [reply]
Yes, that's a possibility for the above error.

PocketBase requires go1.18+ (for Windows you may need go1.19+ because of #6). This is mentioned in the README.
